src/event.rs

```diff
@@ -627,6 +634,28 @@ pub struct KeyboardInput {
     pub modifiers: ModifiersState,
 }
 
+/// Describes an event from input method.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum Ime {
+    /// Notifies when the IME was enabled.
+    Enabled,
```

Comment:
I still don't get why `Ime::Enabled` and `Ime::Disabled` exist, and when the user can expect to receive them - are they just so you know when you don't need to call `Window::set_ime_position`, as an optimization? If so, I feel like there might be a better way to express this (or maybe they're not really warranted).

In any case I'd like for them to be more documented.

Reply:
Yeah, and the user can disable IME on the fly. Also it's some kind of feedback to `set_ime_allowed`.